Code Refactor & Update

robmyers opened the pull request:
Some big changes.

See //FIXME: tags for some queries.

I've moved files and changed their headers so the code layout is easier for WordPress.

I've refactored the code so that all metadata generation is in the class file.

I've made the code that identifies and formats license information more efficient.

I've added extra fields to the media editor (and formatting for the resulting metadata to the class file).

I've replaced the entirely client-side button code with server-side generation for the html the button js uses. The button html, css and js are still works in progress.

BjornW commented:
Hi Rob,

Great to see so much work being done! Thanks!

Just a minor suggestion: I'd prefer many smaller commits than one large
commit. One large commit makes it hard to get an overview of your
changes, while smaller ones provide an insight into your process as well
as the result.

grtz
BjornW


On 13-07-16 00:07, Rob Myers wrote:
> Some big changes.
> 
> See //FIXME: tags for some queries.
> 
> I've moved files and changed their headers so the code layout is easier
> for WordPress.
> 
> I've refactored the code so that all metadata generation is in the class
> file.
> 
> I've made the code that identifies and formats license information more
> efficient.
> 
> I've added extra fields to the media editor (and formatting for the
> resulting metadata to the class file).
> 
> I've replaced the entirely client-side button code with server-side
> generation for the html the button js uses. The button html, css and js
> are still works in progress.
> 
> ------------------------------------------------------------------------
> 
> 
>         You can view, comment on, or merge this pull request online at:
> 
>   https://github.com/creativecommons/wordpress-plugin/pull/11
> 
> 
>         Commit Summary
> 
>   * Re-arrange and re-factor code. Add extra information fields to media
>     eidtor. Add new more server-side version of button code.
> 
> 
>         File Changes
> 
>   * *D* class.creativecommons.php
>     <https://github.com/creativecommons/wordpress-plugin/pull/11/files#diff-0>
>     (921)
>   * *A* creative-commons.php
>     <https://github.com/creativecommons/wordpress-plugin/pull/11/files#diff-1>
>     (22)
>   * *M* css/cc-button.css
>     <https://github.com/creativecommons/wordpress-plugin/pull/11/files#diff-2>
>     (93)
>   * *D* inc/save-image.php~
>     <https://github.com/creativecommons/wordpress-plugin/pull/11/files#diff-3>
>     (138)
>   * *A* includes/ccbutton.php
>     <https://github.com/creativecommons/wordpress-plugin/pull/11/files#diff-4>
>     (98)
>   * *R* includes/class.creativecommons.php
>     <https://github.com/creativecommons/wordpress-plugin/pull/11/files#diff-5>
>     (441)
>   * *A* includes/save-image.php
>     <https://github.com/creativecommons/wordpress-plugin/pull/11/files#diff-6>
>     (221)
>   * *A* js/cc-button-support.js
>     <https://github.com/creativecommons/wordpress-plugin/pull/11/files#diff-7>
>     (318)
>   * *D* js/cc-button.js
>     <https://github.com/creativecommons/wordpress-plugin/pull/11/files#diff-8>
>     (481)
>   * *D* plugin.php
>     <https://github.com/creativecommons/wordpress-plugin/pull/11/files#diff-9>
>     (72)
>   * *D* plugin.php~
>     <https://github.com/creativecommons/wordpress-plugin/pull/11/files#diff-10>
>     (88)
>   * *D* save-image.php
>     <https://github.com/creativecommons/wordpress-plugin/pull/11/files#diff-11>
>     (205)
>   * *M* widgets/creativecommons_widget.php
>     <https://github.com/creativecommons/wordpress-plugin/pull/11/files#diff-12>
>     (12)
> 
> 
>         Patch Links:
> 
>   * https://github.com/creativecommons/wordpress-plugin/pull/11.patch
>   * https://github.com/creativecommons/wordpress-plugin/pull/11.diff
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/creativecommons/wordpress-plugin/pull/11>, or mute
> the thread
> <https://github.com/notifications/unsubscribe/AAD-EvmSoeaTMrqMtGOOgTOE8PMbGmoMks5qVBAsgaJpZM4JK4HE>.
> 


-- 

met vriendelijke groet,
Bjorn Wijers

* b u r o b j o r n .nl *
digitaal vakmanschap | digital craftsmanship

Postbus 14145
3508 SE Utrecht
The Netherlands

tel: +31 6 49 74 78 70
http://www.burobjorn.nl

robmyers commented:
I'll make sure to do that in future. :-)

On Thu, Jul 14, 2016 at 3:51 AM, BjornW <notifications@github.com> wrote:

> Hi Rob,
>
> Great to see so much work being done! Thanks!
>
> Just a minor suggestion: I'd prefer many smaller commits than one large
> commit. One large commit makes it hard to get an overview of your
> changes, while smaller ones provide an insight into your process as well
> as the result.
>
> grtz
> BjornW
>
>
> On 13-07-16 00:07, Rob Myers wrote:
> > Some big changes.
> >
> > See //FIXME: tags for some queries.
> >
> > I've moved files and changed their headers so the code layout is easier
> > for WordPress.
> >
> > I've refactored the code so that all metadata generation is in the class
> > file.
> >
> > I've made the code that identifies and formats license information more
> > efficient.
> >
> > I've added extra fields to the media editor (and formatting for the
> > resulting metadata to the class file).
> >
> > I've replaced the entirely client-side button code with server-side
> > generation for the html the button js uses. The button html, css and js
> > are still works in progress.
> >
> > ------------------------------------------------------------------------
> >
> >
> > You can view, comment on, or merge this pull request online at:
> >
> > https://github.com/creativecommons/wordpress-plugin/pull/11
> >
> >
> > Commit Summary
> >
> > * Re-arrange and re-factor code. Add extra information fields to media
> > eidtor. Add new more server-side version of button code.
> >
> >
> > File Changes
> >
> > * *D* class.creativecommons.php
> > <
> https://github.com/creativecommons/wordpress-plugin/pull/11/files#diff-0>
> > (921)
> > * *A* creative-commons.php
> > <
> https://github.com/creativecommons/wordpress-plugin/pull/11/files#diff-1>
> > (22)
> > * *M* css/cc-button.css
> > <
> https://github.com/creativecommons/wordpress-plugin/pull/11/files#diff-2>
> > (93)
> > * *D* inc/save-image.php~
> > <
> https://github.com/creativecommons/wordpress-plugin/pull/11/files#diff-3>
> > (138)
> > * *A* includes/ccbutton.php
> > <
> https://github.com/creativecommons/wordpress-plugin/pull/11/files#diff-4>
> > (98)
> > * *R* includes/class.creativecommons.php
> > <
> https://github.com/creativecommons/wordpress-plugin/pull/11/files#diff-5>
> > (441)
> > * *A* includes/save-image.php
> > <
> https://github.com/creativecommons/wordpress-plugin/pull/11/files#diff-6>
> > (221)
> > * *A* js/cc-button-support.js
> > <
> https://github.com/creativecommons/wordpress-plugin/pull/11/files#diff-7>
> > (318)
> > * *D* js/cc-button.js
> > <
> https://github.com/creativecommons/wordpress-plugin/pull/11/files#diff-8>
> > (481)
> > * *D* plugin.php
> > <
> https://github.com/creativecommons/wordpress-plugin/pull/11/files#diff-9>
> > (72)
> > * *D* plugin.php~
> > <
> https://github.com/creativecommons/wordpress-plugin/pull/11/files#diff-10>
> > (88)
> > * *D* save-image.php
> > <
> https://github.com/creativecommons/wordpress-plugin/pull/11/files#diff-11>
> > (205)
> > * *M* widgets/creativecommons_widget.php
> > <
> https://github.com/creativecommons/wordpress-plugin/pull/11/files#diff-12>
> > (12)
> >
> >
> > Patch Links:
> >
> > * https://github.com/creativecommons/wordpress-plugin/pull/11.patch
> > * https://github.com/creativecommons/wordpress-plugin/pull/11.diff
> >
> > —
> > You are receiving this because you are subscribed to this thread.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/creativecommons/wordpress-plugin/pull/11>, or mute
> > the thread
> > <
> https://github.com/notifications/unsubscribe/AAD-EvmSoeaTMrqMtGOOgTOE8PMbGmoMks5qVBAsgaJpZM4JK4HE
> >.
> >
>
>
> --
>
> met vriendelijke groet,
> Bjorn Wijers
>
> * b u r o b j o r n .nl *
> digitaal vakmanschap | digital craftsmanship
>
> Postbus 14145
> 3508 SE Utrecht
> The Netherlands
>
> tel: +31 6 49 74 78 70
> http://www.burobjorn.nl
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/creativecommons/wordpress-plugin/pull/11#issuecomment-232631471>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AABU8nyWSS2bKeieDjygGC3fkMjM-wkQks5qVhSagaJpZM4JK4HE>
> .
>

BjornW commented:
Awesome!

grtz
BjornW

On 15-07-16 19:12, Rob Myers wrote:
> I'll make sure to do that in future. :-)
> 
> On Thu, Jul 14, 2016 at 3:51 AM, BjornW <notifications@github.com> wrote:
> 
>> Hi Rob,
>>
>> Great to see so much work being done! Thanks!
>>
>> Just a minor suggestion: I'd prefer many smaller commits than one large
>> commit. One large commit makes it hard to get an overview of your
>> changes, while smaller ones provide an insight into your process as well
>> as the result.
>>
>> grtz
>> BjornW
>>
>>
>> On 13-07-16 00:07, Rob Myers wrote:
>> > Some big changes.
>> >
>> > See //FIXME: tags for some queries.
>> >
>> > I've moved files and changed their headers so the code layout is easier
>> > for WordPress.
>> >
>> > I've refactored the code so that all metadata generation is in the class
>> > file.
>> >
>> > I've made the code that identifies and formats license information more
>> > efficient.
>> >
>> > I've added extra fields to the media editor (and formatting for the
>> > resulting metadata to the class file).
>> >
>> > I've replaced the entirely client-side button code with server-side
>> > generation for the html the button js uses. The button html, css and js
>> > are still works in progress.
>> >
>> > ------------------------------------------------------------------------
>> >
>> >
>> > You can view, comment on, or merge this pull request online at:
>> >
>> > https://github.com/creativecommons/wordpress-plugin/pull/11
>> >
>> >
>> > Commit Summary
>> >
>> > * Re-arrange and re-factor code. Add extra information fields to media
>> > eidtor. Add new more server-side version of button code.
>> >
>> >
>> > File Changes
>> >
>> > * *D* class.creativecommons.php
>> > <
>> https://github.com/creativecommons/wordpress-plugin/pull/11/files#diff-0>
>> > (921)
>> > * *A* creative-commons.php
>> > <
>> https://github.com/creativecommons/wordpress-plugin/pull/11/files#diff-1>
>> > (22)
>> > * *M* css/cc-button.css
>> > <
>> https://github.com/creativecommons/wordpress-plugin/pull/11/files#diff-2>
>> > (93)
>> > * *D* inc/save-image.php~
>> > <
>> https://github.com/creativecommons/wordpress-plugin/pull/11/files#diff-3>
>> > (138)
>> > * *A* includes/ccbutton.php
>> > <
>> https://github.com/creativecommons/wordpress-plugin/pull/11/files#diff-4>
>> > (98)
>> > * *R* includes/class.creativecommons.php
>> > <
>> https://github.com/creativecommons/wordpress-plugin/pull/11/files#diff-5>
>> > (441)
>> > * *A* includes/save-image.php
>> > <
>> https://github.com/creativecommons/wordpress-plugin/pull/11/files#diff-6>
>> > (221)
>> > * *A* js/cc-button-support.js
>> > <
>> https://github.com/creativecommons/wordpress-plugin/pull/11/files#diff-7>
>> > (318)
>> > * *D* js/cc-button.js
>> > <
>> https://github.com/creativecommons/wordpress-plugin/pull/11/files#diff-8>
>> > (481)
>> > * *D* plugin.php
>> > <
>> https://github.com/creativecommons/wordpress-plugin/pull/11/files#diff-9>
>> > (72)
>> > * *D* plugin.php~
>> > <
>> https://github.com/creativecommons/wordpress-plugin/pull/11/files#diff-10>
>> > (88)
>> > * *D* save-image.php
>> > <
>> https://github.com/creativecommons/wordpress-plugin/pull/11/files#diff-11>
>> > (205)
>> > * *M* widgets/creativecommons_widget.php
>> > <
>> https://github.com/creativecommons/wordpress-plugin/pull/11/files#diff-12>
>> > (12)
>> >
>> >
>> > Patch Links:
>> >
>> > * https://github.com/creativecommons/wordpress-plugin/pull/11.patch
>> > * https://github.com/creativecommons/wordpress-plugin/pull/11.diff
>> >
>> > —
>> > You are receiving this because you are subscribed to this thread.
>> > Reply to this email directly, view it on GitHub
>> > <https://github.com/creativecommons/wordpress-plugin/pull/11>, or mute
>> > the thread
>> > <
>>
> https://github.com/notifications/unsubscribe/AAD-EvmSoeaTMrqMtGOOgTOE8PMbGmoMks5qVBAsgaJpZM4JK4HE
>> >.
>> >
>>
>>
>> --
>>
>> met vriendelijke groet,
>> Bjorn Wijers
>>
>> * b u r o b j o r n .nl *
>> digitaal vakmanschap | digital craftsmanship
>>
>> Postbus 14145
>> 3508 SE Utrecht
>> The Netherlands
>>
>> tel: +31 6 49 74 78 70
>> http://www.burobjorn.nl
>>
>> —
>> You are receiving this because you modified the open/close state.
>> Reply to this email directly, view it on GitHub
>>
> <https://github.com/creativecommons/wordpress-plugin/pull/11#issuecomment-232631471>,
>> or mute the thread
>>
> <https://github.com/notifications/unsubscribe/AABU8nyWSS2bKeieDjygGC3fkMjM-wkQks5qVhSagaJpZM4JK4HE>
>> .
>>
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/creativecommons/wordpress-plugin/pull/11#issuecomment-233011238>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAD-EhBkpucf_La7ZUcKlxtLQee6U8MYks5qV7-UgaJpZM4JK4HE>.
> 


-- 

met vriendelijke groet,
Bjorn Wijers

* b u r o b j o r n .nl *
digitaal vakmanschap | digital craftsmanship

Postbus 14145
3508 SE Utrecht
The Netherlands

tel: +31 6 49 74 78 70
http://www.burobjorn.nl

brylie commented:
@all-contributors please add @rheaplex for code
